### PR TITLE
[release-4.10] OCPBUGS-2746: Fixes SNAT-ing Logic for EgressIPs

### PR DESCRIPTION
**Pull request by tssurya**
Backport of https://github.com/openshift/ovn-kubernetes/pull/1334

This is almost like a new PR! Had too many conflicts, took like 3hrs and I had to abandon the unit tests cause they seem broken on 4.10. So reviewers please review with care.

### CODE REVIEW

File: go-controller/hybrid-overlay/pkg/util/util.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func StartNodeWatch(h types.NodeHandler, wf *factory.WatchFactory) {
 			node := obj.(*kapi.Node)
 			h.Delete(node)
 		},
-	}, nil)
+	}, nil, wf.GetHandlerPriority(""))
```
Comment (trozet) [inline go-controller/hybrid-overlay/pkg/util/util.go:81]: note looks like StartNodeWatch isnt even used

**Review by trozet — APPROVED** [review]
/lgtm
/label backport-risk-assessed

**Comment by dcbw** [comment]
/override ci/prow/e2e-aws-ovn-local-to-shared-gateway-mode-migration
GRPC slow alerts are not this PR

**Pull request merged**